src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java

```diff
@@ -498,8 +500,19 @@ public AboutToStartOrSubmitCallbackResponse addCaseNumberSubmitted(
         @RequestBody CallbackRequest callbackRequest
     ) {
         Map<String, Object> caseDataUpdated = callbackRequest.getCaseDetails().getData();
+        CaseData caseData = CaseUtils.getCaseData(callbackRequest.getCaseDetails(), objectMapper);
+        List<CaseEventDetail> eventsForCase = caseEventService.findEventsForCase(String.valueOf(caseData.getId()));
+
+        Optional<String> previousState = eventsForCase.stream()
+            .map(CaseEventDetail::getStateId)
+            .findFirst();
+        log.info("State during the fl401 add case number in addCaseNumberSubmitted: {}", previousState.get());
+        caseDataUpdated.put("isAddCaseNumberAdded", previousState.get().equalsIgnoreCase(String.valueOf(SUBMITTED_PAID)) ? Yes : No);
+        log.info("fl401 add case number flag: {}", caseDataUpdated.get("isAddCaseNumberAdded"));
```

Comment:
refactor this code.

We dont need caseData variable.
Alter the condition for 510
Verify previousState isPresent.